File: robogamer/src/bots/EnginePower.ts
export default class EnginePower {
    public static readonly NoPower = new EnginePower();

    public readonly left: number; // from -1.0 to 1.0
    public readonly right: number; // from -1.0 to 1.0

    public constructor(left: number = 0, right:number = 0){
        this.left = Math.min(Math.max(left, -1), 1);
        this.right = Math.min(Math.max(right, -1), 1);
    }

    
}

File: putlebot-server/src/bots/engine_power.rs
#[derive(Debug, Clone, Copy)]
pub struct EnginePower{
    left: f64, // from -1.0 to 1.0
    right: f64, // from -1.0 to 1.0
}

impl EnginePower {
    pub fn new(left: f64, right: f64) -> EnginePower {
        EnginePower{
            left: left.max(-1.0).min(1.0),
            right: right.max(-1.0).min(1.0),
        }
    }

    pub fn left(self) -> f64 {
        self.left
    }

    pub fn right(self) -> f64 {
        self.right
    }
}